core/cmd/remote_client_test.go

```diff
@@ -568,6 +568,39 @@ func TestClient_GetConfiguration(t *testing.T) {
 	assert.Equal(t, cp.EnvPrinter.SessionTimeout, cfg.SessionTimeout())
 }
 
+// This is a hacky way to capture stdout
+// TODO: consider implementing something that can be easily
+// mocked in tests:
+// https://app.shortcut.com/chainlinklabs/story/46809/make-stdout-testable
+func captureStdout(run func()) string {
+	realStdout := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	run()
+
+	w.Close()
+	captured, _ := ioutil.ReadAll(r)
+	os.Stdout = realStdout
+	return string(captured)
+}
+
+func TestClient_ConfigDump(t *testing.T) {
```

Comment:
It's technically possible to implement this test as an Example as well. It needs some changes in the helper functions and I don't like that we have to hardcode the output instead of comparing it to the source.

Reply:
I agree that an example could be messy, but surely we don't have to capture std out, do we? IMHO that should be absolute last resort since it comes with all kinds of complications and serializes everything. Don't we already have a pattern for this?

Reply:
If there is a way to do it differently, please let me know. In some other cases we're using a Renderer, which can be easily mocked. It would be nice to pipe all output through a Renderer of some kind - I opened a story to figure that out: https://app.shortcut.com/chainlinklabs/story/46809/make-stdout-testable

Reply:
Refactored to get a string before it's printed to stdout.